tests/compiler/test_typing.py

```diff
@@ -143,18 +145,72 @@ def test_cast_to_number(snippetcompiler):
     root: Namespace = scopes.get_child("__config__")
     a = root.lookup("a").get_value()
     b = root.lookup("b").get_value()
+    c = root.lookup("b").get_value()
     x = root.lookup("x").get_value()
     y = root.lookup("y").get_value()
     z = root.lookup("z").get_value()
     u = root.lookup("u").get_value()
     assert Number().validate(a)
     assert Number().validate(b)
+    assert Number().validate(c)
     assert Number().validate(x)
     assert Number().validate(y)
     assert Number().validate(z)
     assert Number().validate(u)
 
 
+def test_compare_float_int_attribute(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+    entity Test:
+        float i = 0
+    end
+
+    entity Test2:
+        int i = 0
+    end
+
+    implement Test using std::none
+    implement Test2 using std::none
+
+    val1 = Test(i = 42.0)
+    val2 = Test2(i = 42)
+    val2.i = val1.i
+    val1.i = val2.i
+    """,
+    )
+    (_, scopes) = compiler.do_compile()
+
+
+def test_assign_float_to_int(snippetcompiler):
+    snippetcompiler.setup_for_error(
+        """
+    entity Test:
+        int i = 0
+    end
+    implement Test using std::none
+    Test(i = 42.1)
+        """,
+        "Could not set attribute `i` on instance `__config__::Test (instantiated at {dir}/main.cf:6)` "
+        "(reported in Construct(Test) ({dir}/main.cf:6))\n"
+        "caused by:\n"
+        "  Invalid value '42.1', expected int (reported in Construct(Test) ({dir}/main.cf:6))",
+    )
+
+
+def test_assign_int_to_float(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+    entity Test:
+        float i = 0
+    end
+    implement Test using std::none
+    Test(i = 42)
```

Comment:
Is this 42 or 42.0? I think that assertion should be part of the test, e.g. as an `isinstance` check.